File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.java
class Solution {
    public int maximumLengthSubstring(String s) {
             int max_len = 0;
        int i =0;
        int j =0;

        HashMap<Character, Integer> count = new HashMap<>();

        while(j < s.length()){
            count.put(s.charAt(j), count.getOrDefault(s.charAt(j), 0) + 1);

            while(count.get(s.charAt(j)) > 2){
                count.put(s.charAt(i), count.getOrDefault(s.charAt(i), 1) - 1);
                i++;
            }

            max_len = Math.max(max_len,(j-i+1));

            j++;
        }
        return max_len;   
    }
}

File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        i = 0
        j = 0
        max_len = 0
        freq = {}

        while(j < len(s)):
            freq[s[j]] = freq.get(s[j], 0) + 1

            while(freq.get(s[j],0) > 2):
                freq[s[i]] -=1
                i+=1

            max_len = max(max_len,(j-i+1))
            j+=1
        return max_len